### PR TITLE
Add RD-270, RD301, LR129

### PR DESCRIPTION
**Pull request by Capkirk123**
Add the LR-129 from BDB, RD-270/M from CH-4, and clone RD-120 for RD-301

### CODE REVIEW

File: GameData/ROEngines/PartConfigs/RD301_RE.cfg [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	MODEL
+	{
+		model = ROEngines/Assets/RealEngines/RD120
+		scale = 0.64, 0.64, 0.64
+	}
+	scale = 0.64
+
+	node_stack_top = 0.0, 0.7122473, 0.0, 0.0, 1.0, 0.0
+	node_stack_bottom = 0.0, -0.9684762, 0.0, 0.0, -1.0, 0.0, 0
+	node_attach = 0.0, 1.116298, 0.0, 0.0, 1.0, 0.0, 4
```
Comment (DRVeyl) [inline GameData/ROEngines/PartConfigs/RD301_RE.cfg:19]: This is somewhat a weird way to do this, using all of MODEL.scale, part.scale, and part.rescaleFactor.  I certainly don't know what to expect!  Can we strip all three out [implying the default stock rescaleFactor of not-1... 0.625 I think?], set the attach node positions correctly for that, and then just correctly set rescaleFactor to where it needs to be?

File: GameData/ROEngines/RealPlume/RD270M_CH4_RealPlume.cfg [hunk 2]
```diff
@@ -0,0 +1,31 @@
+@PART[ROE-RD270M]:FOR[RealPlume]:NEEDS[SmokeScreen,!Waterfall]
+{
```
Comment (DRVeyl) [inline GameData/ROEngines/RealPlume/RD270M_CH4_RealPlume.cfg:2]: ```@PART[ROE-RD270M]:BEFORE[RealPlume]:NEEDS[!Waterfall]```

File: GameData/ROEngines/RealPlume/RD270M_CH4_RealPlume.cfg [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	@MODULE[ModuleEngines*]
+	{
+		%powerEffectName = Pentaborane_Lower
+	}
+	@MODULE[ModuleEngineConfigs]
+	{
+		%type = ModuleEnginesRF
+	}
```
Comment (DRVeyl) [inline GameData/ROEngines/RealPlume/RD270M_CH4_RealPlume.cfg:10]: Delete.  Do not change MEC type in a Plume patch.  (Let RO global patcher fix this up, that's part of its job.)

File: GameData/ROEngines/RealPlume/RD270M_CH4_RealPlume.cfg [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+		flarePosition = 0,0,-0.8
+		flareScale = 0.55
+
+		plumePosition = 0,0,1.0
+		plumeScale = 3
+
+		fumePosition = 0,0,1.5
+		fumeScale = 4.0
+	}
+}
```
Comment (DRVeyl) [inline GameData/ROEngines/RealPlume/RD270M_CH4_RealPlume.cfg:31]: Add newline

File: GameData/ROEngines/RealPlume/RD270_CH4_RealPlume.cfg [hunk 5]
```diff
@@ -0,0 +1,24 @@
+@PART[ROE-RD270]:FOR[RealPlume]:NEEDS[SmokeScreen,!Waterfall]
```
Comment (DRVeyl) [inline GameData/ROEngines/RealPlume/RD270_CH4_RealPlume.cfg:1]: @PART[ROE-RD270]:BEFORE[RealPlume]:NEEDS[!Waterfall]

File: GameData/ROEngines/RealPlume/RD270_CH4_RealPlume.cfg [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    }
+
+    @MODULE[ModuleEngineConfigs]
+    {
+        @CONFIG,*
+        {
+            %powerEffectName = Hypergolic_LowerRed_shock
+        }
+    }
+}
```
Comment (DRVeyl) [inline GameData/ROEngines/RealPlume/RD270_CH4_RealPlume.cfg:24]: Newline before EOF

**Review by DRVeyl — CHANGES_REQUESTED** [review]

File: GameData/ROEngines/PartConfigs/RD301_RE.cfg (hunk 1 above)
Comment (Capkirk123) [reply GameData/ROEngines/PartConfigs/RD301_RE.cfg]: Fixed, uses rescale now.

**Pull request merged**